### PR TITLE
Create new Harvesting guides

### LINKED ISSUE #408: [Guide] Activating delegated harvesting (desktop wallet)
> The guide should be short and direct, including screenshots if necessary. The wallet approach will be shown as the first alternative to announce a transaction (e.g: Method #1: Using the Wallet).

Guide to adapt: https://nemtech.github.io/guides/accountlink/activating-delegated-harvesting.html

### LINKED ISSUE #565: [Guide] Activating delegated harvesting (symbol-bootstrap)
There's a guide to enable delegated harvesting manually:
https://docs.symbolplatform.com/guides/accountlink/activating-delegated-harvesting.html

And there's plans to document how to do it using the Wallet (#408)

We also need to document how to do it using `symbol-bootstrap` which is the simplest method.

### PR DESCRIPTION
**Pull request by segfaultxavi**
Create new folder for them and add guides for local, remote and delegated harvesting using Wallet, Bootstrap, CLI and SDK.
Fixes #565 
Fixes #408

### CODE REVIEW

**Comment by segfaultxavi** [comment]
Blocked because the Harvesting procedure in the Wallet is being changed.

**Comment by segfaultxavi** [comment]
https://github.com/nemtech/catapult-server/issues/153 resolved. Blocked now only because of Desktop Wallet.

File: source/guides/network/running-a-test-net-node.rst [hunk 1]
```diff
@@ -138,7 +140,7 @@ Once the node is running with ``symbol-bootstrap start`` and you have funded its
 
     symbol-bootstrap link
 
-This creates the required :ref:`VrfKeyLinkTransaction <vrf-key-link-transaction>` and submits it to the network. If it succeeds (it might take some seconds, as the transaction needs to be confirmed) your new node is ready to harvest.
+This creates the required :ref:`AccountKeyLink <account-key-link-transaction>` and :ref:`VrfKeyLink <vrf-key-link-transaction>` transactions and announces them to the network. If it succeeds (it might take some seconds, as the transaction needs to be confirmed) your new node is ready to harvest.
 
 .. note:: Without extra parameters, ``symbol-bootstrap link`` tries to send the registration transaction to the local node (running on the other terminal) which will forward it to the rest of the network. If your node is not running at this moment, or it is not an API node, you can provide the URL of a testnet node using ``--url``. Find a `list of nodes here <https://forum.nem.io/t/nem-symbol-0-10-0-release-announcement-testnet-launch/25863>`_.
```
Comment (fboucquez) [inline source/guides/network/running-a-test-net-node.rst:145]: There is a new --useKnownRestGateways in link and enrolSupernode commands that use the best known NGL (highest height) node to announce the transactions. We would keep the list up to date depending on the network type. User doesn't need to have a fully synced local node or manually put an URL to finish up the registration

Comment (segfaultxavi) [reply source/guides/network/running-a-test-net-node.rst]: Thanks, added!

**Comment by segfaultxavi** [comment]
I still want to talk about manually entering the keys instead of using the "Link all keys" button, but that will be dealt in issue #596.

**Pull request merged**